letsencrypt/cli.py

```diff
@@ -341,6 +405,27 @@ def _auth_from_domains(le_client, config, domains):
 
     return lineage
 
+def _avoid_invalidating_lineage(config, lineage, original_server):
+    "Do not renew a valid cert with one from a staging server!"
```

Comment:
I think the code in this function is a bit hacky, but it works.

I'm curious if we can validate the certificate in this function. We require the `ca-certificates` package to be installed which I believe is used by programs like `pip`. If there is a way in Python for us to find the location of the certificates bundle, I believe we can use `fullchain.pem` and `PyOpenSSL` to validate the certificate. I think that might be a better and more reliable test than what we have here.

I don't think we have to block on this though.